### PR TITLE
overseer becomes orchestra

### PR DESCRIPTION
**Pull request by drahnr**
* [ ] move from `metered-channel` to `prioritized-metered-channel`, not part of this PR, required for crates-io publishing, but also #5517  
* [x] renames the proc-macro `overlord` to `orchestra`
* [x] renames all `Overseer` suffixed/prefixed types to be `Orchestra` suffixed/prefixed, i.e. `FromOverseer` -> `FromOrchestra`, `ToOrchestra`, `OrchestraError` etc
* [ ] Decoupling was harder than anticipated, particularly the `trait SpawnNamed` caused some trouble here, which required `struct SpawnGlue<S: SpawnNamed>` being introduced to bridge the `sc_core::traits::SpawnNamed` with the `trait orchestra::Spawner` which are both foreign to `polkadot-overseer`. Currently this type is leaked, but should be eventually be hidden inside the `TestContext` and friends, such that in`polkadot-overseer` only `S: SpawnNamed`  exists and are the converted internally (I leave this open what it exactly means, at this point in time within the `OverseerGen` and helper `fn`s). This is an _integration_ beauty issue and not so much a reason to delay the merge of this PR.

### CODE REVIEW

**Review by tdimitrov — APPROVED** [review]

File: node/overseer/orchestra/proc-macro/src/impl_builder.rs [hunk 1]
```diff
@@ -83,7 +83,7 @@ pub(crate) fn impl_builder(info: &OverseerInfo) -> proc_macro2::TokenStream {
 
 	// Helpers to use within quote! macros
 	let spawner_where_clause: syn::TypeParam = parse_quote! {
-			S: #support_crate ::SpawnNamed + Send
+			S: #support_crate ::Spawner
```
Comment (vstakhov) [inline node/overseer/orchestra/proc-macro/src/impl_builder.rs:86]: So we have finally removed that `Send` requirement.

File: node/overseer/orchestra/proc-macro/src/impl_builder.rs [hunk 2]
```diff
@@ -455,7 +455,7 @@ pub(crate) fn impl_builder(info: &OverseerInfo) -> proc_macro2::TokenStream {
 				// explicitly assure the required traits are implemented
 				fn trait_from_must_be_implemented<E>()
 				where
-					E: std::error::Error + Send + Sync + 'static + From<#support_crate ::OverseerError>
+					E: std::error::Error + Send + Sync + 'static + From<#support_crate ::OrchestraError>
```
Comment (vstakhov) [inline node/overseer/orchestra/proc-macro/src/impl_builder.rs:458]:
```suggestion
					E: ::std::error::Error + Send + Sync + 'static + From<#support_crate ::OrchestraError>
```

File: node/overseer/orchestra/proc-macro/src/impl_builder.rs [hunk 3]
```diff
@@ -706,13 +706,13 @@ pub(crate) fn impl_task_kind(info: &OverseerInfo) -> proc_macro2::TokenStream {
...
-		) -> ::std::result::Result<OverseenSubsystem<M>, #error_ty >
+		) -> ::std::result::Result<OrchestratedSubsystem<M>, #error_ty >
 		where
-			S: #support_crate ::SpawnNamed,
+			S: #support_crate ::Spawner,
 			M: std::fmt::Debug + Send + 'static,
 			TK: TaskKind,
 			Ctx: #support_crate ::SubsystemContext<Message=M>,
-			E: std::error::Error + Send + Sync + 'static + From<#support_crate ::OverseerError>,
+			E: std::error::Error + Send + Sync + 'static + From<#support_crate ::OrchestraError>,
```
Comment (vstakhov) [inline node/overseer/orchestra/proc-macro/src/impl_builder.rs:715]:
```suggestion
			E: ::std::error::Error + Send + Sync + 'static + ::std::convert::From<#support_crate ::OrchestraError>,
```

File: node/overseer/orchestra/src/lib.rs [hunk 4]
```diff
@@ -14,29 +14,31 @@
...
 
-//! # Overseer
+//! # Orchestra
 //!
-//! `overseer` implements the Overseer architecture described in the
-//! [implementers-guide](https://w3f.github.io/parachain-implementers-guide/node/index.html).
-//! For the motivations behind implementing the overseer itself you should
-//! check out that guide, documentation in this crate will be mostly discussing
-//! technical stuff.
+//! `orchestra` provides a global information flow of what a token of information.
```
Comment (vstakhov) [inline node/overseer/orchestra/src/lib.rs:19]: nit: maybe it is a good idea to mention that it was previously called `overseer` to help transition?

**Review by vstakhov — APPROVED** [review]
I'd suggest some small neats.

File: node/overseer/orchestra/src/lib.rs (hunk 4 above)
Comment (drahnr) [reply node/overseer/orchestra/src/lib.rs:19]: Technically the `overseer` still exists as `polkadot-overseer` - only the _inner_ naming changed and I wanted to remove the polkadot references from the `orchestra/*` folders.

File: node/overseer/orchestra/proc-macro/src/impl_builder.rs (hunk 1 above)
Comment (drahnr) [reply node/overseer/orchestra/proc-macro/src/impl_builder.rs:86]: It's implicitly required by the `Spawner` trait already.

**Review by sandreim — APPROVED** [review]
Nit: file names still reference `overseer` - `node/overseer/orchestra/proc-macro/src/parse/parse_overseer_struct.rs`. Otherwise LGTM.

File: Cargo.toml [hunk 5]
```diff
@@ -83,8 +83,8 @@ members = [
 	"node/network/gossip-support",
 	"node/network/dispute-distribution",
 	"node/overseer",
-	"node/overseer/overseer-gen",
-	"node/overseer/overseer-gen/proc-macro",
+	"node/overseer/orchestra",
+	"node/overseer/orchestra/proc-macro",
```
Comment (ordian) [inline Cargo.toml:87]: so now we have overseer *and* orchestra :P

**Pull request merged**